File: app/src/main/java/com/smartneck/twofive/Fit/LoginActivity.java
package com.smartneck.twofive.Fit;

import android.Manifest;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.graphics.Color;
import android.os.Bundle;
import android.os.Handler;
import android.support.annotation.Nullable;
import android.support.v4.app.ActivityCompat;
import android.support.v7.app.AppCompatActivity;
import android.text.Editable;
import android.text.TextWatcher;
import android.util.Log;
import android.view.View;
import android.view.WindowManager;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.smartneck.twofive.Fit.Join.Join_1_Activity;
import com.smartneck.twofive.Fit.Main.MainActivity;
import com.smartneck.twofive.Fit.util.Address;
import com.smartneck.twofive.Fit.util.HttpConnect;
import com.smartneck.twofive.Fit.util.Param;
import com.smartneck.twofive.Fit.util.User.User;

import static com.smartneck.twofive.Fit.FindAccountActivity.ACCOUNT;

public class LoginActivity extends AppCompatActivity {
    EditText et_member_id, et_member_password;
    CheckBox cb_auto_login;
    TextView tv_member_join, tv_member_account , tv_member_password;
    Button btn_login;
    Handler handler;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
        init();



    }

    /* 초기화 */
    private void init(){
        handler = new Handler();

        et_member_id = findViewById(R.id.et_member_id);
        et_member_password = findViewById(R.id.et_member_password);
        btn_login = findViewById(R.id.btn_login);
        tv_member_account = findViewById(R.id.tv_member_account);
        tv_member_password = findViewById(R.id.tv_member_password);
        tv_member_join = findViewById(R.id.tv_member_join);
        cb_auto_login = findViewById(R.id.cb_auto_login);
        setEvent();


        /* 블루투스 권한설정 */
        int IS_ACCESS_COARSE_LOCATION = ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION);
        int IS_ACCESS_FINE_LOCATION = ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION);

        if (android.os.Build.VERSION.SDK_INT >= 29){
            if (IS_ACCESS_FINE_LOCATION == PackageManager.PERMISSION_GRANTED) {
            } else {
                ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.ACCESS_FINE_LOCATION}, 2);
            }
        }else{
            if (IS_ACCESS_COARSE_LOCATION == PackageManager.PERMISSION_GRANTED && IS_ACCESS_FINE_LOCATION == PackageManager.PERMISSION_GRANTED) {
            } else {
                ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.ACCESS_COARSE_LOCATION}, 1);
                ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.ACCESS_FINE_LOCATION}, 2);
            }
        }
    }
    private void setEvent(){
        et_member_password.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {

            }

            @Override
            public void afterTextChanged(Editable s) {
                if (et_member_id.length() >= 8 && et_member_password.length() > 0) {
//                    btn_login.setBackgroundResource(R.drawable.button_background_enabled);
                    btn_login.setBackgroundColor(Color.parseColor("#cc1b17"));

                    btn_login.setTextColor(Color.parseColor("#ffffff"));
                } else {
//                    btn_login.setBackgroundResource(R.drawable.edit_text_background);
                    btn_login.setBackgroundColor(Color.parseColor("#bdbdbd"));
                    btn_login.setTextColor(Color.parseColor("#ffffff"));
                }
            }
        });

        tv_member_join.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Log.d(TAG, "----- tv_member_join onClick");

                Intent intent = new Intent(getApplicationContext(), Join_1_Activity.class);
                startActivity(intent);
            }
        });

        tv_member_account.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Intent intent = new Intent(getApplicationContext() , FindAccountActivity.class);
                startActivity(intent);
            }
        });
        tv_member_password.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext() , com.smartneck.twofive.Fit.FindPasswordActivity.class);
                startActivity(intent);
            }
        });

        btn_login.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Log.d(TAG, "----- btn_login onClick");
                String id = et_member_id.getText().toString().trim();
                String pw = et_member_password.getText().toString().trim();
                login(id, pw);

            }
        });
    }


    private void login(final String id, final String pw) {

        User.setAutologinState(this, cb_auto_login.isChecked());

        new Thread(new Runnable() {
            @Override
            public void run() {
                final HttpConnect httpConnect = new HttpConnect();
                Param param = new Param();
                param.add("id", id);
                param.add("pw", pw);
                final Address address = new Address();
                if (httpConnect.httpConnect(param.getParam(), address.getLogin()) == 200) {
                    if (!httpConnect.getReceiveMessage().equals("fail")) {
//                        User.MemberNo = Integer.parseInt(httpConnect.getReceiveMessage());
                        handler.post(new Runnable() {
                            @Override
                            public void run() {
                                User.getUserInfoJson(getApplicationContext(), httpConnect.getReceiveMessage());
                                Intent intent = new Intent(getApplicationContext(), MainActivity.class);
                                startActivity(intent);
                                finish();
                            }
                        });

                    } else {
                        handler.post(new Runnable() {
                            @Override
                            public void run() {
                                Toast.makeText(com.smartneck.twofive.Fit.LoginActivity.this, getString(R.string.toast_enter_check), Toast.LENGTH_SHORT).show();

                            }
                        });
                    }
                } else {
                    handler.post(new Runnable() {
                        @Override
                        public void run() {
                            Toast.makeText(com.smartneck.twofive.Fit.LoginActivity.this, "internet connect check", Toast.LENGTH_SHORT).show();
                            Log.d(TAG, "run: "+address.getLogin());

                        }
                    });
                }
                Log.d(TAG, "response code: " + httpConnect.getResponseCode());
                Log.d(TAG, "receive message: " + httpConnect.getReceiveMessage());
            }
        }).start();

    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        Toast.makeText(this, "", Toast.LENGTH_SHORT).show();
        Log.d(TAG, "onActivityResult: ");
        if (resultCode == RESULT_OK){
            Log.d(TAG, "onActivityResult: " + data.getStringExtra("account"));
            et_member_id.setText(data.getStringExtra("account"));
        }

    }

    @Override
    protected void onResume() {
        super.onResume();

        if (ACCOUNT != null){
            et_member_id.setText(ACCOUNT);
            ACCOUNT = null;
        }
    }
}

File: app/src/main/java/com/ko/smartneck/medical/twofive/Member/LoginActivity.kt
package com.ko.smartneck.medical.twofive.Member

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.os.Handler
import android.support.v7.app.AppCompatActivity
import android.text.Editable
import android.text.TextWatcher
import android.util.Log
import android.view.WindowManager.LayoutParams
import android.widget.Toast
import android.widget.Toast.LENGTH_SHORT
import com.ko.smartneck.medical.twofive.FindAccountActivity
import com.ko.smartneck.medical.twofive.FindAccountActivity.Companion.ACCOUNT
import com.ko.smartneck.medical.twofive.FindPasswordActivity
import com.ko.smartneck.medical.twofive.GlobalApplication.userPreference
import com.ko.smartneck.medical.twofive.Main.BleConnectActivity
import com.ko.smartneck.medical.twofive.Member.MemberSelectActivity.admin
import com.ko.smartneck.medical.twofive.R
import com.ko.smartneck.medical.twofive.SQ.AdminSQ
import com.ko.smartneck.medical.twofive.WebViewActivity
import com.ko.smartneck.medical.twofive.util.Address
import com.ko.smartneck.medical.twofive.util.Constants.TAG
import com.ko.smartneck.medical.twofive.util.HttpConnect
import com.ko.smartneck.medical.twofive.util.NetworkStatus
import com.ko.smartneck.medical.twofive.util.Param
import kotlinx.android.synthetic.main.activity_login.*
import java.text.SimpleDateFormat
import java.util.*

class LoginActivity : AppCompatActivity() {

    internal var handler: Handler? = null

    companion object{
        lateinit var localAdmin: AdminSQ
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        window.addFlags(LayoutParams.FLAG_KEEP_SCREEN_ON)
        init()

    }


    private fun init() {

        handler = Handler()
        setEvent()
        cb_auto_login.isChecked = true
    }

    private fun setEvent() {

        tv_member_account.setOnClickListener {
            val intent = Intent(applicationContext , FindAccountActivity::class.java)
            startActivity(intent)
        }
        tv_member_password.setOnClickListener {
            val intent = Intent(applicationContext , FindPasswordActivity::class.java)
            startActivity(intent)
        }
        et_member_password!!.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {}
            override fun afterTextChanged(s: Editable?) {
                if (et_member_id!!.length() >= 8 && et_member_password!!.length() > 0) {

                    btn_login!!.setBackgroundColor(Color.parseColor("#cc1b17"))
                    btn_login!!.setTextColor(Color.parseColor("#ffffff"))
                } else {

                    btn_login!!.setBackgroundColor(Color.parseColor("#bdbdbd"))
                    btn_login!!.setTextColor(Color.parseColor("#ffffff"))
                }
            }
        })

        tv_member_join!!.setOnClickListener {
            val intent = Intent(applicationContext, JoinActivity::class.java)
            intent.putExtra("type", "M")
            startActivity(intent)

        }


        tv_password_change.setOnClickListener {
            val intent = Intent(applicationContext , AdminPasswordChangeActivity::class.java)
            startActivity(intent)
        }


        btn_login!!.setOnClickListener {
            val account = et_member_id!!.text.toString().trim { it <= ' ' }
            val password = et_member_password!!.text.toString().trim { it <= ' ' }


            if (NetworkStatus.CURRENT_STATUS != NetworkStatus.TYPE_NOT_CONNECTED){

                val currentDateTime = Calendar.getInstance().time
                val dateFormat = SimpleDateFormat("yyyy-MM-dd", Locale.KOREA).format(currentDateTime)

                val httpConnect = HttpConnect()
                val param = Param()
                param.add("account" , account)
                param.add("password" , password)
                Thread{
                    if (httpConnect.httpConnect(param.value , Address().login , true ) == 200){
                        if (httpConnect.getReceiveMessage() == "success"){
                            val intent = Intent(applicationContext , BleConnectActivity::class.java)
                            if (userPreference.join(account , password , true , dateFormat)){
                                Log.d(TAG , "auto check ${cb_auto_login.isChecked}")

                                userPreference.setBoolean("auto" , cb_auto_login.isChecked)
                                userPreference.setString("autoAccount" , account)
                                userPreference.setString("autoPassword" , password)
                            }
                            admin = userPreference.login(account,password)

                            startActivity(intent)
                            finish()



                        }else{
                            handler!!.post{
                                localLogin(account , password)
                            }

                        }
                    }
                }.start()


            }else{
                localLogin(account , password)

            }

        }
    }

    private fun localLogin(account: String, password: String) {
        admin = userPreference.login(account , password)

        if (admin.account == ""){
            Toast.makeText(this , getString(R.string.toast_enter_check) , LENGTH_SHORT).show()

            return
        }

        val intent = Intent(applicationContext , BleConnectActivity::class.java)

            userPreference.setBoolean("auto" , cb_auto_login.isChecked)
            userPreference.setString("autoAccount" , account)
            userPreference.setString("autoPassword" , password)

        startActivity(intent)
        finish()
    }



//    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
//        super.onActivityResult(requestCode, resultCode, data)
//        Toast.makeText(this, "", Toast.LENGTH_SHORT).show()
//        Log.d(TAG, "onActivityResult: ")
//        if (resultCode == Activity.RESULT_OK) {
//            Log.d(TAG, "onActivityResult: " + data!!.getStringExtra("account"))
//            et_member_id!!.setText(data.getStringExtra("account"))
//        }
//    }

    override fun onResume() {
        super.onResume()

        if (ACCOUNT != null) {
            et_member_id!!.setText(ACCOUNT)
            ACCOUNT = null
        }
    }

    override fun onStop() {
        super.onStop()
    }


}